api/environment.go

```diff
@@ -5,6 +5,10 @@ import (
 	"encoding/json"
 )
 
+// SystemEnvironmentID is the ID of the system environment.
+// This is used for tenant deployment pipeline where environment doesn't matter across stages.
```

Comment:
Does that mean a single stage can contain change to different environments?

What's the use case for this? If it's just the deployment config does not contain the environment info. then there is another option, if we discover we find instances from different environments selected in the same stage according to the deployment config, we throw an error.

Assigning a fake environment to Stage likely to add more complexity.

Reply:
I prefer 2 because it won't introduce an extra state. We should be able to make the UX to smooth the environment selection.